File: src/main/java/com/example/inventory_backend/service/SupplierService.java
package com.example.inventory_backend.service;

import com.example.inventory_backend.entity.Supplier;
import com.example.inventory_backend.repository.SupplierRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class SupplierService {

    private final SupplierRepository supplierRepository;

    public SupplierService(SupplierRepository supplierRepository) {
        this.supplierRepository = supplierRepository;
    }

    public Supplier createSupplier(Supplier supplier) {
        return supplierRepository.save(supplier);
    }

    public List<Supplier> getAllSuppliers() {
        return supplierRepository.findAll();
    }

    public Optional<Supplier> getSupplierById(Long id) {
        return supplierRepository.findById(id);
    }

    public Supplier updateSupplier(Long id, Supplier updatedSupplier) {
        return supplierRepository.findById(id)
                .map(supplier -> {
                    supplier.setName(updatedSupplier.getName());
                    supplier.setEmail(updatedSupplier.getEmail());
                    supplier.setPhone(updatedSupplier.getPhone());
                    supplier.setAddress(updatedSupplier.getAddress());
                    return supplierRepository.save(supplier);
                })
                .orElseThrow(() -> new RuntimeException("Supplier not found"));
    }

    public void deleteSupplier(Long id) {
        supplierRepository.deleteById(id);
    }
}

File: frontend/src/services/supplierService.js
import api from "./api.js";

const API_URL = "/suppliers";

export const getSuppliers = () => api.get(API_URL);

export const getSupplierById = (id) => api.get(`${API_URL}/${id}`);

export const createSupplier = (supplier) =>
  api.post(API_URL, supplier);

export const updateSupplier = (id, supplier) =>
  api.put(`${API_URL}/${id}`, supplier);

export const deleteSupplier = (id) =>
  api.delete(`${API_URL}/${id}`);